### ISSUE TITLE
SLOW Render kernel time

### ISSUE THREAD

**Issue by simbleau** [linebender/vello#163]
I originally thought #161 was the cause of the slowness, but apparently not.
Even after #162 PR the render kernel stage is very slow, unsure why.

GhostScript Tiger, GeForce 1060 3GB:
```
parsing time: 19.066854ms
flattening and encoding time: 1.534247ms
elapsed = 162.079181ms
Element kernel time: 4.987ms
Tile allocation kernel time: 1.812ms
Coarse path kernel time: 2.253ms
Backdrop kernel time: 1.775ms
Binning kernel time: 0.080ms
Coarse raster kernel time: 10.996ms
Render kernel time: 134.673ms
```

**Comment by raphlinus** [comment]
Yeah these numbers are unreasonably slow, about two orders of magnitude slower than expected. On fine rasterization some of it may be #155 but even aside from that these numbers don't look good.

I will want to take a closer look at some point (I have a laptop 1060) but am underwater just at the moment, so make no promise when.

**Comment by raphlinus** [comment]
Ok, trying this on the 1060. It's definitely #155, as reducing the blend stack size speeds that up considerably. I'm not sure the results are as good as I'd want them to be, but I haven't dug deeply yet. I'm pretty sure the driver has changed, as this is not consistent with results I was seeing before on the same hardware, but it's possible my memory is hazy. In any case, I know I'll want to fix blend memory (#156) and also will do some tinkering to see whether any of the other recent changes (gradients, blending, etc) might have caused a regression in fine rasterization performance on this device.

Other than that, your numbers suggest that the GPU clock is really low. Ideally you can set that to a medium-high but stable value. In any case, it should be possible to use the Performance tab in Task Manager (assuming Windows here?) to see the value.

**Comment by raphlinus** [comment]
Closing this, as the blend stack changes have long since landed, and none of the recent performance measurement work is suggesting that there is an issue with rendering performance.
